diff --git a/python/sglang/srt/utils/profile_utils.py b/python/sglang/srt/utils/profile_utils.py
index 1f44af979..850b46618 100644
--- a/python/sglang/srt/utils/profile_utils.py
+++ b/python/sglang/srt/utils/profile_utils.py
@@ -12,6 +12,7 @@ from sglang.srt.managers.io_struct import ProfileReqOutput
 from sglang.srt.model_executor.forward_batch_info import ForwardMode
 from sglang.srt.server_args import get_global_server_args
 from sglang.srt.utils import is_npu
+from sglang.srt.utils import kernel_shape_profiler
 
 _is_npu = is_npu()
 if _is_npu:
@@ -60,6 +61,7 @@ class ProfileManager:
         merge_profiles: bool,
         profile_prefix: str,
         profile_stages: Optional[List[str]] = None,
+        shape_discovery: bool = False,
     ):
         # not supported yet
         assert start_step is None
@@ -86,7 +88,7 @@ class ProfileManager:
             num_steps=num_steps,
             interesting_stages=profile_stages or ["prefill", "decode"],
         )
-
+        self.shape_discovery = shape_discovery
         return ProfileReqOutput(success=True, message="Succeeded")
 
     def manual_start(self):
@@ -281,12 +283,21 @@ class _ProfilerTorch(_ProfilerConcreteBase):
                 else torch_npu.profiler.tensorboard_trace_handler(self.output_dir)
             ),
         )
+
+        # Enable shape metadata for Triton & FlashInfer kernels
+        if self.record_shapes and self.shape_discovery:
+            kernel_shape_profiler.enable()
+
         self.torch_profiler.start()
 
     def stop(self):
         Path(self.output_dir).mkdir(parents=True, exist_ok=True)
 
         self.torch_profiler.stop()
+
+        # Disable kernel shape profiling instrumentation
+        if self.record_shapes and self.shape_discovery:
+            kernel_shape_profiler.disable()
         if not _is_npu:
             # Build filename with only non-zero ranks to maintain backward compatibility
             filename_parts = [self.profile_id, f"TP-{self.tp_rank}"]
