diff --git a/python/sglang/srt/entrypoints/http_server.py b/python/sglang/srt/entrypoints/http_server.py
index 8f7722311a..9351e14caa 100644
--- a/python/sglang/srt/entrypoints/http_server.py
+++ b/python/sglang/srt/entrypoints/http_server.py
@@ -1040,6 +1040,8 @@ async def start_profile_async(obj: Optional[ProfileReqInput] = None):
         merge_profiles=obj.merge_profiles,
         profile_prefix=obj.profile_prefix,
         profile_stages=obj.profile_stages,
+        detailed_annotations=obj.detailed_annotations,
+        shape_discovery=obj.shape_discovery,
     )
     return Response(
         content="Start profiling.\n",
