diff --git a/python/sglang/srt/server_args.py b/python/sglang/srt/server_args.py
index c7162c16d5..d4458092fb 100644
--- a/python/sglang/srt/server_args.py
+++ b/python/sglang/srt/server_args.py
@@ -1972,6 +1972,10 @@ class ServerArgs:
     enable_profile_cuda_graph: A[bool, "Enable profiling of cuda graph capture."] = (
         False
     )
+    enable_shape_discovery_for_cuda_graph_profile: A[
+        bool,
+        "Enable shape discovery during cuda graph capture profiling.",
+    ] = False
     enable_cudagraph_gc: A[
         bool,
         "Enable garbage collection during CUDA graph capture. If disabled (default), GC is frozen during capture to speed up the process.",
