diff --git a/python/sglang/srt/server_args.py b/python/sglang/srt/server_args.py
index 3022bae40..6d30e068b 100644
--- a/python/sglang/srt/server_args.py
+++ b/python/sglang/srt/server_args.py
@@ -526,6 +526,7 @@ class ServerArgs:
     disable_cuda_graph: bool = False
     disable_cuda_graph_padding: bool = False
     enable_profile_cuda_graph: bool = False
+    enable_shape_discovery_for_cuda_graph_profile: bool = False
     enable_cudagraph_gc: bool = False
     enable_layerwise_nvtx_marker: bool = False
     enable_nccl_nvls: bool = False
@@ -3915,6 +3916,11 @@ class ServerArgs:
             action="store_true",
             help="Enable profiling of cuda graph capture.",
         )
+        parser.add_argument(
+            "--enable-shape-discovery-for-cuda-graph-profile",
+            action="store_true",
+            help="Enable shape discovery for cuda graph profiling.",
+        )
         parser.add_argument(
             "--enable-cudagraph-gc",
             action="store_true",
