diff --git a/python/sglang/srt/managers/tokenizer_communicator_mixin.py b/python/sglang/srt/managers/tokenizer_communicator_mixin.py
index e5d42bed8..3d44dd839 100644
--- a/python/sglang/srt/managers/tokenizer_communicator_mixin.py
+++ b/python/sglang/srt/managers/tokenizer_communicator_mixin.py
@@ -327,6 +327,8 @@ class TokenizerCommunicatorMixin:
         merge_profiles: bool = False,
         profile_prefix: Optional[str] = None,
         profile_stages: Optional[List[str]] = None,
+        shape_discovery: bool = False,
+        detailed_annotations: bool = False,
     ):
         self.auto_create_handle_loop()
         env_with_stack: bool = get_bool_env_var("SGLANG_PROFILE_WITH_STACK", "true")
@@ -348,6 +350,8 @@ class TokenizerCommunicatorMixin:
             merge_profiles=merge_profiles,
             profile_prefix=profile_prefix,
             profile_stages=profile_stages,
+            shape_discovery=shape_discovery,
+            detailed_annotations=detailed_annotations,
         )
         return await self._execute_profile(req)
 
